src/VisualStudio/Core/Def/AnalyzerDependency/AnalyzerFileWatcherService.cs

```diff
@@ -110,7 +109,9 @@ internal void TrackFilePathAndReportErrorIfChanged(string filePath, ProjectId pr
                     {
                         if (currentFileUpdateTime != assemblyUpdatedTime)
                         {
-                            RaiseAnalyzerChangedWarning(projectId, filePath);
+                            using var argsBuilder = TemporaryArray<DiagnosticsUpdatedArgs>.Empty;
+                            AddAnalyzerChangedWarningArgs(ref argsBuilder.AsRef(), projectId, filePath);
+                            _updateSource.RaiseDiagnosticsUpdated(argsBuilder.ToImmutableAndClear());
```

Comment:
>_updateSource.RaiseDiagnosticsUpdated(argsBuilder.ToImmutableAndClear()); [](http://example.com/codeflow?start=28&length=73)

check size?

Reply:
➡️ `RaiseDiagnosticsUpdated` checks the count itself